Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,866 @@
+          if ([val isEqualToString: originalId])
+            {
+              [attr setStringValue: @"-1"];
+            }
+        }
+      END_FOR_IN(cons);
+    }
+}
+
+- (void) processStoryboard: (NSXMLDocument *)storyboardXml
+{
+  NSArray *docNodes = [storyboardXml nodesForXPath: @"document" error: NULL];
+
+  if ([docNodes count] > 0)
+    {
+      NSXMLElement *docNode = [docNodes objectAtIndex: 0];
+      NSArray *array = [docNode nodesForXPath: @"//scene" error: NULL];
+      
+      // Set initial view controller...
+      ASSIGN(_initialViewControllerId, [[docNode attributeForName: @"initialViewController"] stringValue]);             
+      FOR_IN(NSXMLElement*, e, array) 
+        {
+          NSXMLElement *doc = [[NSXMLElement alloc] initWithName: @"document"];
+          NSArray *children = [e children];
+          NSXMLDocument *document = nil;
+          NSString *sceneId = [[e attributeForName: @"sceneID"] stringValue]; 
+          NSString *controllerId = nil;
+
+          // Move children...
+          FOR_IN(NSXMLElement*, child, children)
+            {
+              if ([[child name] isEqualToString: @"point"] == YES)
+                continue; // go on if it's a point element, we don't use that in the app...
+              
+              NSArray *subnodes = [child nodesForXPath: @"//application" error: NULL];
+              NSXMLNode *appNode = [subnodes objectAtIndex: 0];
+              [self processChild: child
+                         withDoc: doc
+                     withAppNode: appNode
+                         sceneId: sceneId];
+              
+              // fix other custom objects
+              document = [[NSXMLDocument alloc] initWithRootElement: doc]; // put it into the document, so we can use Xpath.
+              controllerId = [self controllerIdWithDocument: document];
+              RELEASE(doc);
+
+              NSArray *customObjects = [document nodesForXPath: @"//objects/customObject" error: NULL];
+              FOR_IN(NSXMLElement*, coel, customObjects)
+                {
+                  [self processCustomElement: coel
+                                 forDocument: document];
+                }
+              END_FOR_IN(customObjects);
+
+              // Create document...
+              [_scenesMap setObject: document
+                             forKey: sceneId];
+              
+              // Map controllerId's to scenes...
+              if (controllerId != nil)
+                {
+                  [_controllerMap setObject: sceneId
+                                     forKey: controllerId];
+                }
+              
+              RELEASE(document);
+            }
+          END_FOR_IN(children);
+        }
+      END_FOR_IN(array);
+    }
+  else
+    {
+      NSLog(@"No document element in storyboard file");
+    }
+}
+
+- (NSString *) findControllerIdInDocument: (NSXMLDocument *)xmlIn
+                              withObjects: (NSXMLElement *)objects
```

Comment:
You should be using singular for this parameter.

Reply:
It's passing in the "objects" section of the XML, that's why it's plural.